org.opentosca.bus.management.api.osgievent/src/org/opentosca/bus/management/api/osgievent/MBEventHandler.java

```diff
@@ -157,8 +222,518 @@ public void handleEvent(final Event event) {
+            this.getDeployableSubgraph(validNodes, nodeIds2situationIds);
+        for (AbstractRelationshipTemplate relations : topology.getRelationshipTemplates()) {
+            if (deployableAndValidNodeSet.contains(relations.getSource())
+                & deployableAndValidNodeSet.contains(relations.getTarget())) {
+                validRelations.add(relations);
+            }
+        }
+
+        return new ServiceTemplateInstanceConfiguration(deployableAndValidNodeSet, validRelations);
+    }
+
+    private Collection<AbstractNodeTemplate> getDeployableSubgraph(Collection<AbstractNodeTemplate> nodeTemplates,
+                                                                   Map<String, Collection<Long>> nodeIds2situationIds) {
+        Set<AbstractNodeTemplate> validDeploymentSubgraph = new HashSet<AbstractNodeTemplate>(nodeTemplates);
+        Collection<AbstractNodeTemplate> toRemove = new HashSet<AbstractNodeTemplate>();
+
+        for (AbstractNodeTemplate nodeTemplate : nodeTemplates) {
+            Collection<AbstractRelationshipTemplate> hostingRelations = this.getOutgoingHostedOnRelations(nodeTemplate);
+            if (!hostingRelations.isEmpty()) {
+                // if we have hostedOn relations check if it is valid under the situation and is in the set
+                boolean foundValidHost = false;
+                for (AbstractRelationshipTemplate relationshipTemplate : hostingRelations) {
+                    AbstractNodeTemplate hostingNode = relationshipTemplate.getTarget();
+                    if (this.isValidUnderSituations(hostingNode, nodeIds2situationIds)) {
+                        if (nodeTemplates.contains(hostingNode)) {
+                            foundValidHost = true;
+                        }
+                    }
+                    if (foundValidHost) {
+                        // if atleast one hosting relationship is fulfilled we are happy
+                        break;
+                    }
+                }
+                if (!foundValidHost) {
+                    toRemove.add(nodeTemplate);
+                }
+            }
+        }
+
+        if (toRemove.isEmpty()) {
+            return validDeploymentSubgraph;
+        } else {
+            validDeploymentSubgraph.removeAll(toRemove);
+            return getDeployableSubgraph(validDeploymentSubgraph, nodeIds2situationIds);
+        }
+    }
+
+    private boolean isValidUnderSituations(AbstractNodeTemplate nodeTemplate,
+                                           Map<String, Collection<Long>> nodeIds2situationIds) {
+
+
+
+        // check if the situation of the policy is active
+        Collection<Long> situationIds = null;
+
+        if ((situationIds = nodeIds2situationIds.get(nodeTemplate.getId())) == null) {
+            return true;
+        }
+        boolean isValid = true;
+
+        for (Long sitId : situationIds) {
+            isValid &= this.isSituationActive(sitId);
+        }
+
+        return isValid;
```

Comment:
collapse to: `return situationIds.stream().allMatch(this::isSituationActive);`

Reply:
Nope, nobody can read that